x-pack/legacy/plugins/siem/public/components/timeline/timeline_context.tsx

```diff
@@ -4,8 +4,41 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import * as React from 'react';
+import React, { createContext, useContext, useEffect, memo, useState } from 'react';
 
-export const TimelineContext = React.createContext<boolean>(false);
+const initTimelineContext = false;
+export const TimelineContext = createContext<boolean>(initTimelineContext);
+export const useTimelineContext = () => useContext(TimelineContext);
 
-export const TimelineWidthContext = React.createContext<number>(0);
+const initTimelineWidth = 0;
+export const TimelineWidthContext = createContext<number>(initTimelineWidth);
+export const useTimelineWidthContext = () => useContext(TimelineWidthContext);
+
+interface ManageTimelineContextProps {
+  children: React.ReactNode;
+  loading: boolean;
+  width: number;
+}
+
+// To DO we need to refactor this as more complex context/reducer with useReducer
```

Comment:
optional mega ultra nit 3000: if we grep the code for `todo`, the `To DO` in this comment won't show up 😉